bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java

```diff
@@ -127,6 +127,10 @@ public int getTag() {
 
     @Override
     public String toString() {
+        if (resolving) {
+            return BLangConstants.CYCLIC_UNION;
+        }
+        resolving = true;
```

Comment:
Needs to be discussed. Printing error messages for cyclic types (`stackoverflow`) , what would be the best approach?